spec/lib/requirements/backdate_checker_spec.rb

```diff
@@ -4,19 +4,19 @@
   describe "#pre_submit_issues" do
     it "returns no issues if there are none" do
       date = Time.current.change(day: 1, month: 1, year: 2019)
-      issues = Requirements::BackdateChecker.new(date).pre_submit_issues
+      issues = subject.pre_update_issues(date)
```

Comment:
Oh - when did subjects start appear in the tests? We have some precedent for `described_class` but until now I wasn't aware this was being used in the tests - seems it was introduced with fields and is now spreading?

Why did this get introduced now without any prior discussion? I remember it was an area of contention in the early days which I am imagine you are also aware of.

Reply:
Aha, so I wasn't aware it was contentious. I do remember a discussion around this and having all tests do their own setup, execution and checks in a self-contained way. Since we now have a fair few `let`s, it didn't seem like an issue anymore.

Looking at the specs, it does seem to be something I've introduced recently. We can look at reverting it (and doing more typing 😢) if you can help me understand why we should avoid it. [This article](https://thoughtbot.com/blog/lets-not) looks familiar, but is a little extreme. We seem to be in a state of some inconsistency here, and perhaps we should try and write down want we do/n't want, or think about some linting.